process/coordinator/process.go

```diff
@@ -772,12 +782,24 @@ func (tc *transactionCoordinator) processCompleteMiniBlock(
 			log.Debug("RevertToSnapshot", "error", errAccountState.Error())
 		}
 
+		tc.revertProcessedTxsResults(processedTxs)
```

Comment:
You should call this only: if len(processedTxs) > 0, as method ProcessMiniBlock could return nil, err

Reply:
done.